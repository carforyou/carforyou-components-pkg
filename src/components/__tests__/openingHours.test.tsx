import React from "react"
import { render } from "@testing-library/react"

import OpeningHours from "../openingHours"

describe("<Opening hours />", () => {
  const labels = {
    monday: "Monday",
    tuesday: "Tuesday",
    wednesday: "Wednesday",
    thursday: "Thursday",
    friday: "Friday",
    saturday: "Saturday",
    sunday: "Sunday",
    open: "open",
    closed: "closed",
  }

  it("renders correct markup - 24h open and the clock icon", () => {
    const openingHours = [
      {
        open: {
          dayOfWeek: "Sunday",
          time: "00:00",
        },
        close: null,
      },
    ]

    const { container } = render(
      <OpeningHours openingHours={openingHours} labels={labels} clockIcon />
    )

    expect(container).toMatchSnapshot()
  })

  it("renders correct markup - open working days", () => {
    const openingHours = [
      {
        open: {
          dayOfWeek: "Monday",
          time: "08:00",
        },
        close: {
          dayOfWeek: "Monday",
          time: "19:00",
        },
      },
      {
        open: {
          dayOfWeek: "Tuesday",
          time: "08:00",
        },
        close: {
          dayOfWeek: "Tuesday",
          time: "19:00",
        },
      },
      {
        open: {
          dayOfWeek: "Wednesday",
          time: "08:00",
        },
        close: {
          dayOfWeek: "Wednesday",
          time: "19:00",
        },
      },
      {
        open: {
          dayOfWeek: "Thursday",
          time: "08:00",
        },
        close: {
          dayOfWeek: "Thursday",
          time: "19:00",
        },
      },
      {
        open: {
          dayOfWeek: "Friday",
          time: "08:00",
        },
        close: {
          dayOfWeek: "Friday",
          time: "19:00",
        },
      },
      {
        open: {
          dayOfWeek: "Saturday",
          time: "08:00",
        },
        close: {
          dayOfWeek: "Saturday",
          time: "19:00",
        },
      },
    ]

    const { container } = render(
      <OpeningHours openingHours={openingHours} labels={labels} />
    )

    expect(container).toMatchSnapshot()
  })

  it("renders correct markup - closed during lunch time", () => {
    const openingHours = [
      {
        open: {
          dayOfWeek: "Monday",
          time: "07:30",
        },
        close: {
          dayOfWeek: "Monday",
          time: "12:00",
        },
      },
      {
        open: {
          dayOfWeek: "Monday",
          time: "13:30",
        },
        close: {
          dayOfWeek: "Monday",
          time: "18:00",
        },
      },
      {
        open: {
          dayOfWeek: "Tuesday",
          time: "07:30",
        },
        close: {
          dayOfWeek: "Tuesday",
          time: "12:00",
        },
      },
      {
        open: {
          dayOfWeek: "Tuesday",
          time: "13:30",
        },
        close: {
          dayOfWeek: "Tuesday",
          time: "18:00",
        },
      },
      {
        open: {
          dayOfWeek: "Wednesday",
          time: "07:30",
        },
        close: {
          dayOfWeek: "Wednesday",
          time: "12:00",
        },
      },
      {
        open: {
          dayOfWeek: "Wednesday",
          time: "13:30",
        },
        close: {
          dayOfWeek: "Wednesday",
          time: "18:00",
        },
      },
      {
        open: {
          dayOfWeek: "Thursday",
          time: "07:30",
        },
        close: {
          dayOfWeek: "Thursday",
          time: "12:00",
        },
      },
      {
        open: {
          dayOfWeek: "Thursday",
          time: "13:30",
        },
        close: {
          dayOfWeek: "Thursday",
          time: "18:00",
        },
      },
      {
        open: {
          dayOfWeek: "Friday",
          time: "07:30",
        },
        close: {
          dayOfWeek: "Friday",
          time: "12:00",
        },
      },
      {
        open: {
          dayOfWeek: "Friday",
          time: "13:30",
        },
        close: {
          dayOfWeek: "Friday",
          time: "17:00",
        },
      },
      {
        open: {
          dayOfWeek: "Saturday",
          time: "09:00",
        },
        close: {
          dayOfWeek: "Saturday",
          time: "12:00",
        },
      },
    ]

    const { container } = render(
      <OpeningHours openingHours={openingHours} labels={labels} />
    )

    expect(container).toMatchSnapshot()
  })
})
