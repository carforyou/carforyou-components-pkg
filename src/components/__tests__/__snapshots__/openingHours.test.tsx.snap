// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Opening hours /> renders correct markup - 24h open 1`] = `
<div>
  <div
    class="flex mt-15 text-grey-4 text-sm"
  >
    <svg
      height="24"
      viewBox="0 0 22 22"
      width="24"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        stroke="#3696B9"
        stroke-width="2"
        transform="translate(-5 -5)"
      >
        <rect
          height="20"
          rx="10"
          width="20"
          x="6"
          y="6"
        />
        <path
          d="M16 11.2V16l2.4 2.4"
          stroke-linecap="round"
        />
      </g>
    </svg>
    <div
      class="w-12/12 pl-15"
    >
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Mo-Su
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            open 24h
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Opening hours /> renders correct markup - closed during lunch time 1`] = `
<div>
  <div
    class="flex mt-15 text-grey-4 text-sm"
  >
    <svg
      height="24"
      viewBox="0 0 22 22"
      width="24"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        stroke="#3696B9"
        stroke-width="2"
        transform="translate(-5 -5)"
      >
        <rect
          height="20"
          rx="10"
          width="20"
          x="6"
          y="6"
        />
        <path
          d="M16 11.2V16l2.4 2.4"
          stroke-linecap="round"
        />
      </g>
    </svg>
    <div
      class="w-12/12 pl-15"
    >
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Mo-Th
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            07:30-12:00
          </span>
          <span>
            13:30-18:00
          </span>
        </div>
      </div>
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Friday
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            07:30-12:00
          </span>
          <span>
            13:30-17:00
          </span>
        </div>
      </div>
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Saturday
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            09:00-12:00
          </span>
        </div>
      </div>
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Sunday
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            closed
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Opening hours /> renders correct markup - open working days 1`] = `
<div>
  <div
    class="flex mt-15 text-grey-4 text-sm"
  >
    <svg
      height="24"
      viewBox="0 0 22 22"
      width="24"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        stroke="#3696B9"
        stroke-width="2"
        transform="translate(-5 -5)"
      >
        <rect
          height="20"
          rx="10"
          width="20"
          x="6"
          y="6"
        />
        <path
          d="M16 11.2V16l2.4 2.4"
          stroke-linecap="round"
        />
      </g>
    </svg>
    <div
      class="w-12/12 pl-15"
    >
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Mo-Sa
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            08:00-19:00
          </span>
        </div>
      </div>
      <div
        class="w-12/12 flex items-baseline pb-10"
      >
        <div
          class="w-4/12"
        >
          Sunday
        </div>
        <div
          class="ml-20 flex flex-col leading-xs"
        >
          <span>
            closed
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
